pkg/obfuscate/appsec_test.go

```diff
@@ -0,0 +1,251 @@
+package obfuscate
+
+import (
+	"encoding/json"
+	"regexp"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestObfuscateAppSec(t *testing.T) {
+	for _, tc := range []struct {
+		name           string
+		keyRE, valueRE *regexp.Regexp
+		value          string
+		expected       string
+	}{
+		{
+			// The key regexp should take precedence over the value regexp and obfuscate the entire values
+			name:     "sensitive-key",
+			keyRE:    regexp.MustCompile(`k3`),
+			valueRE:  regexp.MustCompile(`SENSITIVE`),
+			value:    `{"triggers":[{"rule_matches":[{"parameters":[{"key_path":[0,1,"k1",2,"k3"],"highlight":["highlighted SENSITIVE value 1","highlighted SENSITIVE value 2","highlighted SENSITIVE value 3"],"value":"the entire SENSITIVE value"}]}]}]}`,
+			expected: `{"triggers":[{"rule_matches":[{"parameters":[{"key_path":[0,1,"k1",2,"k3"],"highlight":["?","?","?"],"value":"?"}]}]}]}`,
+		},
+		{
+			// The key regexp should take precedence over the value regexp and obfuscate the entire values
+			name:     "sensitive-key",
+			keyRE:    regexp.MustCompile(`k3`),
+			valueRE:  nil,
+			value:    `{"triggers":[{"rule_matches":[{"parameters":[{"key_path":[0,1,"k1",2,"k3"],"highlight":["highlighted SENSITIVE value 1","highlighted SENSITIVE value 2","highlighted SENSITIVE value 3"],"value":"the entire SENSITIVE value"}]}]}]}`,
+			expected: `{"triggers":[{"rule_matches":[{"parameters":[{"key_path":[0,1,"k1",2,"k3"],"highlight":["?","?","?"],"value":"?"}]}]}]}`,
+		},
+		{
+			// The key regexp doesn't match and the value regexp does and obfuscates accordingly.
+			name:     "sensitive-value",
+			keyRE:    nil,
+			valueRE:  regexp.MustCompile(`SENSITIVE`),
+			value:    `{"triggers":[{"rule_matches":[{"parameters":[{"key_path":[0,1,"k1",2,"k3"],"highlight":["highlighted SENSITIVE value 1","highlighted value 2","highlighted SENSITIVE value 3"],"value":"the entire SENSITIVE value"}]}]}]}`,
+			expected: `{"triggers":[{"rule_matches":[{"parameters":[{"key_path":[0,1,"k1",2,"k3"],"highlight":["highlighted ? value 1","highlighted value 2","highlighted ? value 3"],"value":"the entire ? value"}]}]}]}`,
+		},
+		{
+			name:     "disabled",
+			keyRE:    nil,
+			valueRE:  nil,
+			value:    `{"triggers":[{"rule_matches":[{"parameters":[{"key_path":[0,1,"k1",2,"k3"],"highlight":["highlighted SENSITIVE value 1","highlighted value 2","highlighted SENSITIVE value 3"],"value":"the entire SENSITIVE value"}]}]}]}`,
+			expected: `{"triggers":[{"rule_matches":[{"parameters":[{"key_path":[0,1,"k1",2,"k3"],"highlight":["highlighted SENSITIVE value 1","highlighted value 2","highlighted SENSITIVE value 3"],"value":"the entire SENSITIVE value"}]}]}]}`,
+		},
+		{
+			name:     "unexpected-json-empty-string",
+			keyRE:    regexp.MustCompile(`SENSITIVE`),
+			valueRE:  regexp.MustCompile(`SENSITIVE`),
+			value:    ``,
```

Comment:
I don't see the difference between this test and the `unexpected-json-empty-value` one that comes right after.
Am I missing something?